app/parse.py

```diff
@@ -8,5 +10,45 @@ class Course:
     duration: str
 
 
+def get_soup(url):
+    content = requests.get(url).content
```

Comment:
Consider adding error handling for the HTTP request to manage potential exceptions, such as network errors or non-200 status codes. This will make the function more robust.